lib/src/graph/identifier.dart

```diff
@@ -183,16 +180,37 @@ class LocalDeclaration extends Identifier implements AbstractDeclaration {
 class HoverResult extends Vertex {
   @override
   String get label => 'hoverResult';
-  String hoverText;
-  HoverResult(this.hoverText);
 
+  /// The documentation string for the element associated to this hover. Can often be `null`.
+  final String docString;
+
+  /// The package for the element associated to this hover. Eg. package:lsif_indexer, dart:core
+  /// Can be `null` (eg. for [LocalDeclaration])
+  final String package;
+
+  /// The declaration for the element associated to this hover.
+  final String declaration;
+
+  HoverResult({@required this.declaration, this.docString, this.package});
+
+  /// Represents the hover result in the following format:
+  /// packageName (eg. `package:lsif_indexer` or `dart:core`)
+  /// -------------------------
+  /// declaration (eg. `final String delclaration`)
+  /// -------------------------
+  /// docString (eg. `/// The declaration for the element associated to this hover.`)
   @override
-  Map<String, Object> toLsif() => {
-        ...super.toLsif(),
-        'result': {
-          'contents': {'kind': 'markdown', 'value': hoverText}
-        }
-      };
+  Map<String, Object> toLsif() {
+    return {
+      ...super.toLsif(),
+      'result': {
+        'contents': [package, declaration, docString]
+            .whereNotNull()
```

Comment:
nit: you could do this as 
`   ... get _hoverTexts [package, declaration, docString];`
then
`  'contents': [for (var text in _hoverTexts) if (text != null) {'language': 'dart', 'value': text}];`
which feels a bit nicer to me - but I really like the collection literal stuff. And then you wouldn't need the extension method.